src/screens/DiscoveryScreenV2/fragment/NewsFragment.js

```diff
@@ -1,140 +1,161 @@
-                navigation.navigate('DetailDomainScreen', {
-                    item: {
-                        id: item.post_id,
-                        score: item?.newsLinkDomain?.credder_score
-                    }
-                })
-            }
-
-            return <RenderItem key={`newsDiscovery-${index}`}
+  const [myId, setMyId] = React.useState('');
+  const [defaultNews] = React.useContext(Context).news;
+
+  const isReady = useIsReady();
+
+  const navigation = useNavigation();
+
+  React.useEffect(() => {
+    const parseToken = async () => {
+      const id = await getUserId();
+      if (id) {
+        setMyId(id);
+      }
+    };
+    parseToken();
+  }, []);
+
+  // React.useEffect(() => {
+  //     if(news.length > 0) setIsFirstTimeOpen(false)
+  // }, [news])
+
+  const renderNewsItem = () => {
+    if (isFirstTimeOpen) {
+      return [<DiscoveryTitleSeparator key="news-title-separator" text="Suggested News" />].concat(
+        defaultNews.news.map((item, index) => {
+          const onContentClicked = () => {
+            navigation.navigate('DetailDomainScreen', {
+              item: {
+                id: item.id,
+                score: item.domain.credderScore,
+                follower: 0
+              }
+            });
+          };
+
+          // Disable on press content if view should be navigated to LinkContextScreen
+          if (!item.dummy)
+            return (
+              <RenderItem
+                key={`news-screen-item-${index}`}
+                item={item}
                 selfUserId={myId}
-                item={contentParam}
-                onPressShare={share.shareNews}
-            // onPressContent={onContentClicked} 
-            />
+                // onPressContent={onContentClicked}
+              />
+            );
         })
+      );
     }
 
-    if (!isReady) return <></>
-
-    if (isLoadingDiscoveryNews) return <View style={styles.fragmentContainer}><LoadingWithoutModal /></View>
-    if (news.length === 0 && !isFirstTimeOpen) return <View style={styles.noDataFoundContainer}>
+    return news.map((item, index) => {
+      const contentParam = newsDiscoveryContentParamBuilder(
+        item.title,
+        item.image,
+        item.description,
+        item.news_url,
+        item?.newsLinkDomain?.logo,
+        item?.newsLinkDomain?.domain_name || item?.site_name,
+        item.createdAt,
+        item.news_link_id,
+        item?.newsLinkDomain?.credder_score,
+        item?.post_id
+      );
+
+      const onContentClicked = () => {
+        navigation.navigate('DetailDomainScreen', {
+          item: {
+            id: item.post_id,
+            score: item?.newsLinkDomain?.credder_score
+          }
+        });
+      };
+
+      return (
+        <RenderItem
+          key={`newsDiscovery-${index}`}
+          selfUserId={myId}
+          item={contentParam}
+          onPressShare={share.shareNews}
+          // onPressContent={onContentClicked}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `onPressContent` prop is commented out in both `RenderItem` components. If this is intentional and you want to disable this functionality, consider removing these lines entirely for cleaner code. If this is unintentional, uncomment these lines to restore the functionality.
```diff
- // onPressContent={onContentClicked}
+ onPressContent={onContentClicked}
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->